Unstructured-IO/unstructured
# feat: script to sync version

- Added script to check/sync versions using `CHANGELOG.md` as a source of truth.
- Script currently only syncs `__version__.py` but can easily be extended to cover other files by adding the files to an array in the script.
- Also updated sphinx `conf.py` to get version dynamically from `__version__.py`

#### Testing:
To check script:
1. Change the version in `__version__.py` to an invalid version.
2. Run `make check-version`.
3. Should fail with an error message indicating valid semver could not be found in `__version__.py`.
4. Run `make version-sync`.
5. Should fail with same error message.
1. Change the version in `__version__.py` to a valid version, that is different from latest version in `CHANGELOG.md`.
2. Run `make check-version`.
3. Should exit 1, reporting diffs that would fix the version in `__version__.py`.
4. Run `make version-sync`.
5. `__version__.py` should now be in sync with `CHANGELOG.md`.
6. Open `CHANGELOG.md` and delete the contents of the file, and save it empty.
7. Run `make check-version`.
8. Should fail with an error message indicating valid semver could not be found in `CHANGELOG.md`.
9. Run `make version-sync`.
10. Should fail with same error message.
11. Undo changes in files.
12. `make check-version` and `make version-sync` should now exit 0 with no changes.

To check sphinx doc version:
1. Add the line `Version |release|` to `docs/source/index.rst`
2. In the `docs` folder, run `make html`
3. Open `docs/build/html/index.html`, version should appear and match the value in `__version__.py`
4. Change the version in `__version__.py`.
2. In the `docs` folder, run `make html`
3. Open `docs/build/html/index.html`, version should still match the value in `__version__.py`
